docs/source/modules/zos_stat.rst

```diff
@@ -0,0 +1,670 @@
+
+    record_format
+      Record format of a data set.
+
+      | **returned**: success
+      | **type**: str
+      | **sample**: VB
+
+    record_length
+      Record length of a data set.
+
+      | **returned**: success
+      | **type**: int
+      | **sample**: 80
+
+    block_size
+      Block size of a data set.
+
+      | **returned**: success
+      | **type**: int
+      | **sample**: 27920
+
+    has_extended_attrs
+      Whether a data set has extended attributes set.
+
+      | **returned**: success
+      | **type**: bool
+      | **sample**:
+
+        .. code-block:: json
+
+            true
+
+    extended_attrs_bits
+      Current values of the EATTR bits for a data set.
+
+      | **returned**: success
+      | **type**: str
+      | **sample**: OPT
+
+    creation_date
+      Date a data set was created.
+
+      | **returned**: success
+      | **type**: str
+      | **sample**: 2025-01-27
+
+    creation_time
+      Time at which a data set was created.
+
+      Only available when a data set has extended attributes.
+
+      | **returned**: success
+      | **type**: str
+      | **sample**: 11:25:52
+
+    expiration_date
+      Expiration date of a data set.
+
+      | **returned**: success
+      | **type**: str
+      | **sample**: 2030-12-31
+
+    last_reference
+      Date where the data set was last referenced.
+
+      | **returned**: success
+      | **type**: str
+      | **sample**: 2025-01-28
+
+    updated_since_backup
+      Whether the data set has been updated since its last backup.
+
+      | **returned**: success
+      | **type**: bool
+
+    jcl_attrs
+      Dictionary containing the names of the JCL job and step that created a data set.
+
+      Only available for data sets with extended attributes.
+
+      | **returned**: success
+      | **type**: dict
+
+      creation_job
+        JCL job that created the data set.
+
+        | **returned**: success
+        | **type**: str
+
+      creation_step
+        JCL job step that created the data set.
+
+        | **returned**: success
+        | **type**: str
+
+
+    volser
+      Name of the volume containing the data set.
+
+      | **returned**: success
+      | **type**: str
+      | **sample**: 000000
+
+    num_volumes
+      Number of volumes where the data set resides.
+
+      | **returned**: success
+      | **type**: int
+      | **sample**: 1
+
+    volumes
+      Names of the volumes where the data set resides.
+
+      | **returned**: success
+      | **type**: list
+      | **elements**: str
+      | **sample**:
+
+        .. code-block:: json
+
+            [
+                "000000",
+                "SCR03"
+            ]
+
+    device_type
+      Generic device type where the data set resides.
+
+      | **returned**: success
+      | **type**: str
+      | **sample**: 3390
+
+    space_units
+      Units used to describe sizes for the data set.
+
+      | **returned**: success
+      | **type**: str
+      | **sample**: TRACK
+
```

Comment:
Sorry but the case thing is jumping out at me.  Also, I believe we used 'trk' for track measure in archive, backup, copy and others that take sizes.